document-store/src/main/java/org/hypertrace/core/documentstore/mongo/MongoCollection.java

```diff
@@ -355,4 +367,13 @@ public boolean bulkUpsert(Map<Key, Document> documents) {
   public void drop() {
     collection.drop();
   }
+
+  // cap the max limit to 10k records
+  private Integer normalizeLimit(Integer limit) {
+    if (limit != null && limit > MAX_LIMIT) {
```

Comment:
Why the max limit? there previously was no max limit (since a cursor doesn't require one for perf reasons) - is there a compelling reason to add one?

Reply:
no specific reason. I remembered we have some 10K limit in Pinot as well. Anyways, we can leave that to the consumers of Mongo

_getting rid of the MAX LIMIT_